fossology/fossology
# feat(DB): Recode copyright tables to UTF-8

## Description

Current encoding FOSSology DB is `SQL_ASCII`. This PR adds a migration script to recode the contents of `copyright`, `keyword` and `ecc` tables to `UTF-8`.

**Note:** This PR also updates the encoding of database to `UTF-8`.

### Changes

1. Add a new migration script to recode `copyright`, `keyword` and `ecc` table contents.
1. Move the UTF-8 filter logic to CPP library.
1. Move the `fo_unicode_clean` to `src/copyright/agent`.
1. Do not run the script if total number of records are `> 500,000` (~15 minutes on 4 vCPU, 16GB vRAM)

## How to test

1. On an old FOSSology, upload files with invalid copyright texts (see #1528 and #1557).
1. Install the branch and run `fo-postinstall`.
    1. The encoding of the DB should be changed to `UTF-8`.
    1. All invalid characters should be removed from the tables.
1. Check if the support script works (`/usr/local/lib/fossology/dbmigrate_change_db_encoding.php`).

Closes #1492
